dspace-spring-rest/src/test/java/org/dspace/app/rest/InitializeEntitiesIT.java

```diff
@@ -0,0 +1,164 @@
+import org.dspace.content.RelationshipType;
+import org.dspace.content.service.EntityTypeService;
+import org.dspace.content.service.RelationshipService;
+import org.dspace.content.service.RelationshipTypeService;
+import org.dspace.services.ConfigurationService;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class InitializeEntitiesIT extends AbstractControllerIntegrationTest {
+
+    @Autowired
+    private RelationshipTypeService relationshipTypeService;
+
+    @Autowired
+    private EntityTypeService entityTypeService;
+
+    @Autowired
+    private RelationshipService relationshipService;
+
+    @Autowired
+    private ConfigurationService configurationService;
+
+    /**
+     * Build the relationships using the standard test XML with the initialize-entities script
+     */
+    @Before
+    public void setup() throws Exception {
+
+        //Set up the database for the next test
+        String pathToFile = configurationService.getProperty("dspace.dir") +
+            File.separator + "config" + File.separator + "entities" + File.separator + "relationship-types.xml";
+        runDSpaceScript("initialize-entities", "-f", pathToFile);
+
+    }
+
+    @After
+    public void destroy() throws Exception {
+        //Clean up the database for the next test
+        context.turnOffAuthorisationSystem();
+        List<RelationshipType> relationshipTypeList = relationshipTypeService.findAll(context);
+        List<EntityType> entityTypeList = entityTypeService.findAll(context);
+        List<Relationship> relationships = relationshipService.findAll(context);
+
+        Iterator<Relationship> relationshipIterator = relationships.iterator();
+        while (relationshipIterator.hasNext()) {
+            Relationship relationship = relationshipIterator.next();
+            relationshipIterator.remove();
+            relationshipService.delete(context, relationship);
+        }
+
+        Iterator<RelationshipType> relationshipTypeIterator = relationshipTypeList.iterator();
+        while (relationshipTypeIterator.hasNext()) {
+            RelationshipType relationshipType = relationshipTypeIterator.next();
+            relationshipTypeIterator.remove();
+            relationshipTypeService.delete(context, relationshipType);
+        }
+
+        Iterator<EntityType> entityTypeIterator = entityTypeList.iterator();
+        while (entityTypeIterator.hasNext()) {
+            EntityType entityType = entityTypeIterator.next();
+            entityTypeIterator.remove();
+            entityTypeService.delete(context, entityType);
+        }
+
+        super.destroy();
+    }
+
+    /**
+     * Verify whether the initialize-entities script can update the relationship types correctly
+     */
+    @Test
+    public void test() throws Exception {
+        List<RelationshipType> relationshipTypes = relationshipTypeService.findAll(context);
```

Comment:
I think this is low priority.  You are technically correct that this test method is testing more than it really needs to (it's repeating some tests performed elsewhere, so it likely doesn't need to verify all 10 relationship types exist in the response).  But, I think this isn't something that should block the code from moving forward.

Reply:
no it is not a blocking issue but this unnecessary slow the test, as other changes are required https://github.com/DSpace/DSpace/pull/2376/files#r280904440

it could be a good opportunity to fix also this minor thing instead than postpone. Anyway, it can be postponed if necessary